### DISCUSSION TITLE
Goto Tab if file is already open, otherwise open a new tab

### DISCUSSION THREAD

**Discussion by omar-shahid** [nvim-tree/nvim-tree.lua#1598] [Q&A] [answered]
Hi,
I am a beginner with Nvim and Lua config, this plugin is awesome for a vscode-like sidebar. The only problem I am facing, which looks like a simple thing that I have missed somewhere, is how to navigate to an open tab if it is already open, else open the file in a new tab. Currently, nvim-tree will open files to a new tab no matter how tabs are opened on that file.

**Comment by alex-courtis** [comment, marked as answer]
> how to navigate to an open tab if it is already open, else open the file in a new tab

As you are new to vim: do you mean a vim tab or a vim window? I am going to assume window.

You can open in a new window split vertcally via `<C-v>` and horizontally via `<C-x>`.

On edit (open) via `<CR>` or `o` or `<2-LeftMouse>` nvim-tree will navigate to the window if it is open.

See [Beginners Guide to Tabs in Vim](https://webdevetc.com/blog/tabs-in-vim/) if you are interested in vim tabs.

Use `<C-t>` to open in a new vim tab. The tab switching behaviour can be customised via options such as `:help nvim-tree.open_on_tab`.

**Reply by omar-shahid** [reply]
Thank you so much for your answer. Yes I'm talking about the former, vim tabs, that has the commands like (`tabnew`, `tabnext` etc). When I try to open an existing opened file, it creates a new tab instead of moving to it.

**Reply by alex-courtis** [reply]
> When I try to open an existing opened file, it creates a new tab instead of moving to it.

I'm not seeing that behaviour with the default config. It sounds like it might be a bug.

Please raise a [bug](https://github.com/kyazdani42/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml) containing your configuration and describe the steps to replicate the issue.

A screen recording would be useful. See https://asciinema.org/

**Reply by omar-shahid** [reply]
Hi,
Apologies I got really confused, the issue is other way around. 

What I want:
- Open the file in a new tab if it is not opened.
- Navigate to the tab if it is opened

What I am getting:
- Nvim opens the file in the current tab, no matter if the file is edited or if it is a different file.

https://user-images.githubusercontent.com/40920737/192084681-de193a92-ccbd-4d80-8559-19d6de59fa6f.mp4

**Reply by alex-courtis** [reply]
> Open the file in a new tab if it is not opened.

That is achieved via `<C-t>` `tabnew`. You _could_ rebind the edit actions `o`, `<CR>` etc. to `tabnew`.

> Navigate to the tab if it is opened

nvim-tree doesn't have that functionality and likely won't ever, as that is not the "vim paradigm" for tabs. See next reply.

**Reply by alex-courtis** [reply]
My opinion:

Your workflow is incorrect: you appear to be trying to use vim as an IDE like VSCode or IntelliJ, with open files as tabs at the top. vim is not an IDE.

Tabs are usually used to separate logical groups of windows and buffers i.e. separate projects. Users will only have a few tabs open.

The usual vim workflow is to have the single (default) tab containing all the window(s) in which the buffers for that project are opened.

To switch between buffers in a window you can use `:help :ls` to view open buffers and `:help :buf` to switch between them. That is completely impractical and most users will use a plugin to manage open buffers and switching between them.

[bufexplorer](https://github.com/jlanzarotta/bufexplorer) is a popular and simple plugin.

nvim-tree is also used to switch buffers in windows. It will use the one opened window by default, however you can open new windows via `<C-x>` and `<C-t>`. When there are multiple open windows nvim-tree will:
* focus the window containing the buffer
* prompt you to pick a window, see `:help nvim-tree.actions.open_file.window_picker`

**Reply by omar-shahid** [reply]
Thanks for the detailed answer and providing additional tips! I appreciate it.
